sag-system/src/main/java/uk/gov/dbt/ndtp/core/SecureAgentGraph.java

```diff
@@ -172,7 +172,18 @@ private static void processConfigPath(YAMLConfigParser ycp, RDFConfigGenerator r
             ConfigStruct configStruct = ycp.runYAMLParser(configPath);
             Model configModel = rcg.createRDFModel(configStruct);
 
-            File rdfConfigPath = File.createTempFile("generated-config-", ".ttl");
+            // Create a secure temp directory and pass it to the temp file
+            String homeDir = System.getProperty("user.home");
+            File secureTempPath = new File(homeDir, ".sag-temp");
+
+            if (!secureTempPath.exists()) {
+                boolean created = secureTempPath.mkdirs();
+                if (!created) {
```

Comment:
You may be able to get away with this 

```suggestion
                if (!secureTempPath.mkdirs()) {
```

Reply:
Nice one, changed in https://github.com/National-Digital-Twin/secure-agent-graph/pull/7/commits/64e62aa2ad79ccbce3da84e17876921c65bacb13